commands/diceRoller.js

```diff
@@ -1,18 +1,67 @@
 let ftl = require('../lib/ftl');
+let { stripContent, getRandomNumber } = require('../lib/utils');
 
-module.exports.register = function (botArguments) {
+module.exports.register = (botArguments) => {
   const bot = botArguments.bot;
   bot.registerCommand('roll', (msg) => {
-    let dice = stripContent(msg.content);
+    let command = stripContent(msg.content);
     try {
-      return rollDecider(dice);
+      return rollDecider(command);
     } catch (err) {
       return err.toString();
     }
   });
 };
 
+// Handles the logic of the command and returns correct messages to chat.
 function rollDecider(command) {
+  validateSubcommands(command); //Checks to make sure all characters are valid
+  let dice = diceDetails(command);
+  let { reroll, add, subtract, amount, sides, modifier } = dice;
+  checkValidroll(amount, add, subtract, sides);
+
+  if (add) {
+    return ftl('roll-output', {
+      amount,
+      sides,
+      modifier: `+ ${modifier}`,
+      result: rollDice(amount, sides, reroll) + modifier,
+    });
+  } else if (subtract) {
+    return ftl('roll-output', {
+      amount,
+      sides,
+      modifier: `- ${modifier}`,
+      result: rollDice(amount, sides, reroll) - modifier,
+    });
+  } else {
+    return ftl('roll-output', {
+      amount,
+      sides,
+      modifier: '',
+      result: rollDice(amount, sides, reroll),
+    });
+  }
+}
+
+// Generates our dice rolls
+function rollDice(amount, sides, reroll) {
+  let diceTotal = 0;
+
+  for (let roll = 0; roll < amount; roll++) {
+    let currentRoll;
+
+    do {
+      currentRoll = getRandomNumber(sides);
+    } while (reroll && currentRoll <= 1);
+
+    diceTotal += currentRoll;
+  }
+  return diceTotal;
+}
+
+// Makes sure all the parts needed to role our dice are present and there are no extranious charactors.
+function validateSubcommands(command) {
```

Comment:
I like pulling the validation out into another step, but I don't think that "subcommand" is the right term to use here. I usually see subcommand as the `fetch` in `git fetch` or the `install` in `npm install`. Here you aren't using that kind of structure, and especially because Eris [has the idea of subcommands](https://abal.moe/Eris/docs/Command#function-registerSubcommand) I think it would be worth avoiding the clashing name.